Apps/fastlane/Fastfile

```diff
@@ -33,41 +33,26 @@ end
 
 # Functions specific to iOS go in the platform block. 
 platform :ios do 
-  # Builds the iOS app and uploads compiled app for internal testing. 
+  # Builds the iOS app and (optionally) uploads compiled app for internal testing. 
   # Build will only be uploaded if the environment variables for uploading are set.
   #
+  # Pre-requisites:
+  # 1. Assumes the build and app version are set in the Xcode project.
+  # 2. Assumes that the CIO workspace environment files are generated (Env.swift, for example). 
+  #
   # Usage: 
-  # `fastlane ios build` <-- if you want to generate a new build number and app version. Common to use for creating unique QA builds. Each build number is guaranteed to be unique.
-  # `fastlane ios build build_number:123 app_version:1.2.3` <-- if you want to specify the build number and app version.   
+  # `fastlane ios build` 
   lane :build do |arguments|    
-    download_ci_code_signing_files
-    
-    new_build_number = arguments[:build_number] || get_new_build_version()
-    new_app_version = arguments[:app_version] || get_new_app_version()
-    build_notes = get_build_notes()
-    test_groups = get_build_test_groups()
-
-    # Modify the source code with the new app version and build number before we compile the iOS app. This is a good idea to do to make installing builds on a test device easier. 
-    # The iOS OS might give errors when trying to install a build of an app if the app is already installed on the device. Having unique build number or app version can avoid those errors. 
-    update_ios_app_versions(
-      build_number: new_build_number,
-      app_version: new_app_version
-    )
-    
-    version_update_script_path = '../../../scripts/update-version.sh'
-    # Since this Fastfile is used by SDK wrapper projects that might not contain this script file, we skip running it if the file does not exist. 
-    if File.exist?(version_update_script_path)
-      UI.important("Updating the SDK's source code version to non-production version. This allows the sample apps to show the SDK version at runtime for app user to better understand the version of the SDK they are running.")
-
-      sh("#{version_update_script_path} \"#{new_app_version}.#{new_build_number}\"")
-    end 
-
-    uses_cocoapods = File.exists?("../Podfile")
-    if uses_cocoapods 
-      UI.message "Project uses CocoaPods. Going to skip SPM dependency downloading."
```

Comment:
To make `ios build` more re-usable, this code was moved into separate functions. 

`ios build` function now has the assumption that before you run it, you have already updated the app version and build number.